juju/juju
# Mechanical changes for Firewall Rules

### Checklist


----

## Description of change

The firewall rules importing task was wrong, most if not all the
importing logic, doesn't validate what's coming in and assumes because
the newly created model is a blank slate, that it can just insert
documents without retry logic.

When making this change to be in line with all the other tasks, I put
the firewallrules type methods back to what they where originally.

In doing these changes I added missing tests for the migration import
tasks, so we can at least verify that our import script works for both
integration and unit level testing.


## QA steps

Ensure all the tests pass.

```sh
juju bootstrap aws test
juju add-model foo
juju set-firewall-rule ssh --whitelist xxx.xxx.xxx.xxx/xxx

juju bootstrap aws other

juju switch test
juju migrate foo other

juju switch foo
```

Open a connection to mongo and ensure that the firewall rule was migrated

```sh
db.firewallRules.find().pretty()
```

Bonus points:
Expose an endpoint after deployment and look in aws console to see the firewall rule is correct opened and migrated.
